### ISSUE TITLE
[BUG] `use-setstate-synchronously` doesn't recognize `assert` statements

### ISSUE THREAD

**Issue by DetachHead** [dart-code-checker/dart-code-metrics#1178]
### Environment and configuration

DCM version: 5.5.1
Dart SDK version:  3.0.0-52.0.dev

<details>
<summary>Configuration</summary>

```yaml
dart_code_metrics:
  extends:
    - package:dart_code_metrics/presets/all.yaml
```
</details>

### What did you do?

```dart
await doThing();
assert(mounted);
setState(() => {}); // error: use-setstate-synchronously
```

### What did you expect to happen?

no error

### What actually happened?

`use-setstate-synchronously` error

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by incendial** [comment]
Oh, considering you have 3.0.0, can you share how is your experience with DCM so far?

**Comment by DetachHead** [comment]
I haven't encountered any problems, although I didn't use it for very long. I was just testing something unrelated and forgot to switch back

**Comment by incendial** [comment]
Still, that's good to know, thank you!

**Comment by incendial** [comment]
@Desdaemon hey, do you want to take a look or should I? If you don't have time, that's totally okay 👍

**Comment by Desdaemon** [comment]
> @Desdaemon hey, do you want to take a look or should I? If you don't have time, that's totally okay 👍 

That'd be great, thanks. I've been having trouble finding free time recently.

**Comment by incendial** [comment]
Available in 5.6.0 🚀
